logics/impls/controller/mod.rs

```diff
@@ -718,27 +694,239 @@ impl<T: Storage<Data>> Internal for T {
         pool: AccountId,
         redeemer: AccountId,
         redeem_amount: Balance,
-        pool_attribute: Option<PoolAttributes>,
+        pool_attributes: Option<PoolAttributesForWithdrawValidation>,
     ) -> Result<()> {
-        let caller_pool = if let Some(_pool_attribute) = pool_attribute {
-            Some((pool, _pool_attribute))
-        } else {
-            None
-        };
-        let (_, shortfall) = self._get_hypothetical_account_liquidity(
-            redeemer,
-            Some(pool),
-            redeem_amount,
-            0,
-            caller_pool,
-        )?;
-        if !shortfall.is_zero() {
-            return Err(Error::InsufficientLiquidity)
-        }
+        if let Some(oracle) = self._oracle() {
```

Comment:
If oracle.unwrap fails, it is just an Err (which basically should not happen), so I want to reduce nesting.
Can we make nesting unnecessary like this?
```rust
let oracle = self._oracle().ok_or(Error::OracleIsNotSet)?;
```

Reply:
I have resolved unnecessary nestings.